frontend/src/utils/cardButtons.js

```diff
@@ -0,0 +1,160 @@
+import {
+    READY,
+    REVIEW_FEEDBACK,
+    COMPLETE,
+    BLOCKED,
+    IN_PROGRESS,
+    IN_REVIEW,
+    TEAM_PERMISSIONS,
+    MANAGE_CARDS,
+    REVIEW_CARDS,
+    TRUSTED_REVIEWER,
+  } from "../constants";
+  
+  export function getTeamPermissions({ authUser, viewedUser }) {
+    let result = {};
+  
+    for (let permission of TEAM_PERMISSIONS) result[permission] = false;
+  
+    if (authUser.isSuperuser) {
+      for (let permission of TEAM_PERMISSIONS) result[permission] = true;
+    } else {
+      // we look at what teams this user belongs to. If authUser has a permission on one of those teams then they have the permission for the user
+      for (let authedTeamId in authUser.permissions.teams) {
+        if (viewedUser.teamMemberships[authedTeamId]) {
+          let heldPermissions =
+            authUser.permissions.teams[authedTeamId].permissions;
+          for (let permission of heldPermissions) {
+            result[permission] = true;
+          }
+        }
+      }
+    }
+    return result;
+  }
+  
+  export function getShowAddReviewButton({ card, permissions, isReviewer }) {
+    const REVIEW_STATUSES = [IN_REVIEW, COMPLETE, REVIEW_FEEDBACK];
+    if (isReviewer && REVIEW_STATUSES.indexOf(card.status) !== -1) return true;
+    if (
+      (permissions[REVIEW_CARDS] || permissions[TRUSTED_REVIEWER]) &&
+      REVIEW_STATUSES.indexOf(card.status) !== -1
+    )
+      return true;
+  
+    return false;
+  }
```

Comment:
Hey Faith, please make the code dryer. There's repetition with:
```
REVIEW_STATUSES.indexOf(card.status) !== -1
```
How about:
```
return (
    (ifYour'reReviwer or
      haveReviewPermission or
      areTrustedReviewer) &&
    REVIEW_STATUSES.indexOf(card.status) !== -1
  )
```
then you can also remove the `return true` and `return false` because they'll be automatically returned from the statement